src/api/context.ts

```diff
@@ -0,0 +1,65 @@
+/*********************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import Command from '@oclif/command'
+import Listr = require('listr')
+import * as os from 'os'
+import * as path from 'path'
+
+import { CHE_OPERATOR_CR_PATCH_YAML_KEY, CHE_OPERATOR_CR_YAML_KEY, LOG_DIRECTORY_KEY } from '../common-flags'
+import { readCRFile } from '../util'
+
+import { KubeHelper } from './kube'
+
+/**
+ * chectl context.
+ */
+export namespace ChectlContext {
+  export const IS_OPENSHIFT = 'isOpenShift'
+  export const IS_OPENSHIFT4 = 'isOpenShift4'
+  export const START_TIME = 'startTime'
+  export const END_TIME = 'endTime'
+  export const CONFIG_DIR = 'configDir'
+  export const ERROR_LOG = 'errorLog'
+  export const COMMAND_ID = 'commandId'
+  export const CUSTOM_CR = 'customCR'
+  export const CR_PATCH = 'crPatch'
+  export const LOGS_DIRECTORY = 'directory'
+
+  const ctx: any = {}
+
+  export async function init(flags: any, command: Command): Promise<void> {
+    const kube = new KubeHelper(flags)
+    ctx[IS_OPENSHIFT] = await kube.isOpenShift()
+    ctx[IS_OPENSHIFT4] = await kube.isOpenShift4()
+
+    ctx.highlightedMessages = [] as string[]
+    ctx[START_TIME] = Date.now()
+    ctx[CUSTOM_CR] = readCRFile(flags, CHE_OPERATOR_CR_YAML_KEY)
+    ctx[CR_PATCH] = readCRFile(flags, CHE_OPERATOR_CR_PATCH_YAML_KEY)
```

Comment:
Same as above. Will it work for Helm installer? What to do with Helm patch?

Reply:
Not everything are moved there.
Just the most common attributes.
So, yes it will works as before.

Reply:
Anyway, it introduces installer specific properties that are never used by another installer.

Reply:
@tolusha could you separate installer specific properties keys in a separate group at least ? And add a comment for them?